fedimint-api-client/src/api/mod.rs

```diff
@@ -645,51 +654,71 @@ pub struct ConnectorRegistryBuilder {
 impl ConnectorRegistryBuilder {
     #[allow(clippy::unused_async)] // Leave room for async in the future
     pub async fn bind(self) -> anyhow::Result<ConnectorRegistry> {
-        let mut inner: BTreeMap<String, JitTryAnyhow<DynConnector>> = BTreeMap::new();
-
-        if self.iroh_enable {
-            let iroh_connector = JitTryAnyhow::new_try(move || async move {
-                Ok(Arc::new(
-                    api::iroh::IrohConnector::new(
-                        self.iroh_dns,
-                        self.iroh_pkarr_dht,
-                        self.iroh_next,
-                    )
-                    .await?,
-                ) as DynConnector)
-            });
-            inner.insert("iroh".to_string(), iroh_connector);
+        // Create initialization functions for each connector type
+        let mut connectors_lazy: BTreeMap<String, (ConnectorInitFn, OnceCell<DynConnector>)> =
+            BTreeMap::new();
+
+        // WS connector init function
+        let builder_ws = self.clone();
+        let ws_connector_init = Arc::new(move || {
+            let builder = builder_ws.clone();
+            Box::pin(async move { builder.build_ws_connector().await })
+                as Pin<Box<dyn Future<Output = anyhow::Result<DynConnector>> + Send>>
+        });
+        connectors_lazy.insert("ws".into(), (ws_connector_init.clone(), OnceCell::new()));
+        connectors_lazy.insert("wss".into(), (ws_connector_init.clone(), OnceCell::new()));
+
+        // Iroh connector init function
+        let builder_iroh = self.clone();
+        connectors_lazy.insert(
```

Comment:
Probably doesn't make too much of a difference either way, but if `iroh_enable` is false, wouldn't it be easier to just not insert the connector like we were doing originally?